lib/src/ctx/context.rs

```diff
@@ -118,43 +134,39 @@ impl<'a> Context<'a> {
 		self.add_deadline(Instant::now() + timeout)
 	}
 
+	/// Add the current transaction to the context, so that it can be fetched
+	/// where necessary, including inside the query planner.
 	pub fn add_transaction(&mut self, txn: Option<&Transaction>) {
-		if let Some(txn) = txn {
-			self.transaction = Some(txn.clone());
-		}
+		self.transaction = txn.map(Clone::clone);
```

Comment:
For future reference, [`Option::cloned`](https://doc.rust-lang.org/std/option/enum.Option.html#method.cloned) is a more idiomatic conversion from `Option<&T>` to `Option<T>`

Reply:
I've replaced this @finnbear in https://github.com/surrealdb/surrealdb/pull/2227/commits/02b4e5d94677111e62aa5a7c1c7c29394f72e81d